codec/test_codec.go

```diff
@@ -84,29 +85,44 @@ type MyInnerStruct3 struct {
 }
 
 type myStruct struct {
-	InnerStruct  MyInnerStruct      `serialize:"true"`
-	InnerStruct2 *MyInnerStruct     `serialize:"true"`
-	Member1      int64              `serialize:"true"`
-	Member2      uint16             `serialize:"true"`
-	MyArray2     [5]string          `serialize:"true"`
-	MyArray3     [3]MyInnerStruct   `serialize:"true"`
-	MyArray4     [2]*MyInnerStruct2 `serialize:"true"`
-	MySlice      []byte             `serialize:"true"`
-	MySlice2     []string           `serialize:"true"`
-	MySlice3     []MyInnerStruct    `serialize:"true"`
-	MySlice4     []*MyInnerStruct2  `serialize:"true"`
-	MyArray      [4]byte            `serialize:"true"`
-	MyInterface  Foo                `serialize:"true"`
-	MySlice5     []Foo              `serialize:"true"`
-	InnerStruct3 MyInnerStruct3     `serialize:"true"`
-	MyPointer    *Foo               `serialize:"true"`
+	InnerStruct  MyInnerStruct              `serialize:"true"`
+	InnerStruct2 *MyInnerStruct             `serialize:"true"`
+	Member1      int64                      `serialize:"true"`
+	Member2      uint16                     `serialize:"true"`
+	MyArray2     [5]string                  `serialize:"true"`
+	MyArray3     [3]MyInnerStruct           `serialize:"true"`
+	MyArray4     [2]*MyInnerStruct2         `serialize:"true"`
+	MySlice      []byte                     `serialize:"true"`
+	MySlice2     []string                   `serialize:"true"`
+	MySlice3     []MyInnerStruct            `serialize:"true"`
+	MySlice4     []*MyInnerStruct2          `serialize:"true"`
+	MyArray      [4]byte                    `serialize:"true"`
+	MyInterface  Foo                        `serialize:"true"`
+	MySlice5     []Foo                      `serialize:"true"`
+	InnerStruct3 MyInnerStruct3             `serialize:"true"`
+	MyPointer    *Foo                       `serialize:"true"`
+	MyMap1       map[string]string          `serialize:"true"`
+	MyMap2       map[int32][]MyInnerStruct3 `serialize:"true"`
+	MyMap3       map[MyInnerStruct2][]int32 `serialize:"true"`
+	MyMap4       map[int32]*int32           `serialize:"true"`
+	MyMap5       map[int32]int32            `serialize:"true"`
```

Comment:
can we also test map with fixed size array values? (do fixed size array keys also work?)

Reply:
Fixed in 77c262f88